virttest/utils_windows/drive.py

```diff
@@ -37,3 +39,68 @@ def get_floppy_drives_letter(session):
     """
     cond = "MediaType!=0 AND MediaType!=11 AND MediaType!=12"
     return _logical_disks(session, cond=cond, props=["DeviceID"])
+
+
+def rescan_disk(session):
+    """
+    Rescan disk in windows guest.
+
+    :param session: Session object.
+    """
+    rescan_cmd = "echo rescan > cmd"
+    rescan_cmd += " && echo exit >> cmd"
+    rescan_cmd += " && diskpart /s cmd"
+    session.cmd(rescan_cmd)
+
+
+def extend_shrink_disk(session, did, disk_letter, operation=""):
+    """
+    Extend/Shrink a partition on disk to desired_size in windows guest.
+    Default extend/shrink to max usage size.
+
+    :param session: Session object.
+    :param did: Disk index which show in 'diskpart list disk'.
+    :param disk_letter: Disk letter to extend or shrink.
+    :param operation: Operation is extend or shrink.
+
+    :return: Desired shrink size(only for shrink action).
+    """
+    list_disk_cmd = "echo list disk > cmd && "
+    list_disk_cmd += "echo exit >> disk && diskpart /s cmd"
+    disks = session.cmd_output(list_disk_cmd, timeout=120)
+
+    for disk in disks.splitlines():
+        if re.search(r"DISK %s" % did, disk, re.I | re.M):
+            cmd_header = 'echo list disk > cmd &&'
+            cmd_header += ' echo select disk %s >> cmd &&' % did
+            cmd_footer = ' echo exit >> cmd && diskpart /s cmd'
+            cmd_footer += ' && del /f cmd'
+            detail_cmd = ' echo detail disk >> cmd &&'
+            detail_cmd = ' '.join([cmd_header, detail_cmd, cmd_footer])
+            logging.debug("Detail for 'Disk%s'" % did)
+            details = session.cmd_output(detail_cmd)
+            volume_id = re.findall("Volume (\d)\s+%s" % disk_letter, details, re.M)[0]
+
+            cmd_header += ' echo select volume %s >> cmd && ' % volume_id
+            extend_cmd = ' echo extend >> cmd &&'
+            extend_cmd = ' '.join([cmd_header, extend_cmd, cmd_footer])
+            if operation == "extend":
+                logging.info("extending the 'Disk%s'" % did)
+                session.cmd(extend_cmd)
+            elif operation == "shrink":
+                shrink_query_cmd = ' echo shrink querymax >> cmd &&'
+                shrink_query_cmd = ' '.join([cmd_header, shrink_query_cmd, cmd_footer])
+                logging.info("quering maximum number of reclaimble for the 'Disk%s'" % did)
+                query_output = session.cmd_output(shrink_query_cmd)
+                logging.info(query_output)
+                maximum_num = re.findall("(\d+) MB", query_output)[0]
+                if int(maximum_num) > 1024:
+                    desired_size = (int(maximum_num)-1024)
```

Comment:
need spaces between `-`.
please setup some pep8 checks on your text editor, so you can avoid this kind of error

Reply:
has updated, thanks remind, I installed pep8, but it's so sorry that not checked this error.